visual-recognition/v3.ts

```diff
@@ -87,75 +86,84 @@ class VisualRecognitionV3 extends GeneratedVisualRecognitionV3 {
     }
     if (
       !(
-        this.xor(params.images_file, params.url) ||
-        this.xor(params.images_file, _parameters.url)
+        this.xor(_images_file, params.url) ||
+        this.xor(_images_file, _parameters.url)
       )
     ) {
-      return _callback(err);
+      throw err;
     }
-    const _url = _parameters.url || params.url;
-    const _classifier_ids = _parameters.classifier_ids || params.classifier_ids;
-    const _owners = _parameters.owners || params.owners;
-    const _threshold = _parameters.threshold || params.threshold;
-    let _obj = {
-      url: _url,
-      classifier_ids: _classifier_ids,
-      owners: _owners,
-      threshold: _threshold
-    };
-    // remove null/undefined keys
-    Object.keys(_obj).forEach(key => {
-      _obj[key] == null && delete _obj[key];
+  }
+
+  private parseParameters(params) {
+    let parameters;
+    try {
+      parameters = JSON.parse(params.parameters);
+    } catch (e) {
+      parameters = {};
+    }
+    const parameters_keys = ['url', 'classifier_ids', 'owners', 'threshold'];
+    let _parameters = {};
+    parameters_keys.forEach(key => {
+      if (parameters[key] || params[key]) {
+        _parameters[key] = parameters[key] || params[key];
+      }
     });
-    return Object.keys(_obj).length > 0 ? _obj : null;
+    return Object.keys(_parameters).length > 0 ? _parameters : null;
   }
 
   classify(params, callback) {
-    const _parameters = this.parseParameters(params, callback) || {};
-    // set defaults for classify()
-    if (!_parameters.classifier_ids) {
-      _parameters.classifier_ids = ['default'];
+    const _callback = typeof callback === 'function' ? callback : () => {};
```

Comment:
What if you use: 
```js
classify(params, callback = Function.prototype)
```

If `callback` is not a function we should throw an error rather than set a default no-op func. 
What do you think?